src/task.rs

```diff
@@ -3,7 +3,50 @@
 use std::pin::Pin;
 
 use futures::prelude::*;
-use futures::task::{Context, Poll};
+use futures::future::FutureObj;
+use futures::task::{Context, Poll, Spawn, SpawnError};
+
+/// A [`Spawn`] handle to runtime's thread pool for spawning futures.
+///
+/// This allows integrating runtime with libraries based on explicitly passed spawners.
+#[derive(Debug)]
+pub struct Spawner {
+    _reserved: (),
+}
+
+impl Spawner {
+    /// Construct a new [`Spawn`] handle.
+    pub fn new() -> Self {
+        Self {
+            _reserved: (),
+        }
+    }
+}
+
+impl Default for Spawner {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl Clone for Spawner {
+    fn clone(&self) -> Self {
+        Self::new()
+    }
+}
+
+impl Spawn for Spawner {
+    fn spawn_obj(&mut self, future: FutureObj<'static, ()>) -> Result<(), SpawnError> {
+        (&*self).spawn_obj(future)
+    }
+}
+
+impl<'a> Spawn for &'a Spawner {
+    fn spawn_obj(&mut self, future: FutureObj<'static, ()>) -> Result<(), SpawnError> {
+        runtime_raw::current_runtime()
+            .spawn_boxed(future.boxed())
```

Comment:
This does have the downside of double-boxing the future.

I wonder if it makes sense to change `Runtime::spawn_boxed` back to `Runtime::spawn_obj` to avoid this.

Reply:
Perhaps. It's worth considering that `spawn_obj` _is_ going to be deprecated at some point though, which prompted the migration in the first place. Not sure what to do about that.

There's also #39 to use `smallbox` which may be of interest